keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1813 @@
+        the scheduler, and also for configuring cgroup settings.  
+
+  * **Rows 13 - 16**
+
+    ![](./pod-request-limit-set.png)
+
+      * When both pod-level request and limit are explicitly specified,the scheduler and
+        cgroups will use those values directly.
+    
+      * Rows 13, 15 Annotation [c]: When container-level limit is not set, the
+        pod-level limit is applied to each container's cgroup maximum value. This
+        is because container-level limit is implied in this case, and some
+        runtimes (like the Java runtime) rely on container-level cgroup maximum
+        values for fine tuning their components.
+<br>
+
+##### Clarifying Tricky Cases and Design Decisions with Pod Level Resources
+
+1. [Invalid Pod Spec] Row 5 from the scenarios table: Pod Limits Defined; Pod
+   Requests & Container Specs Unspecified.
+   
+   ![](./example-1.png)
+  
+   * The Challenge of Missing Requests: In this case, Kubernetes cannot determine
+     the appropriate pod-level request as pod-level request is not set explicitly,
+     and since container-level requests are also not set, there's no way for
+     kubernetes to derive pod-level requests from the resource spec.
+   * Prioritizing Explicit Resource Requests: Instead of relying on potentially
+     inaccurate default values or guessing the user's intent, this KEP proposes
+     rejecting such a pod specification. This encourages users to explicitly
+     define their resource requests, promoting clarity and preventing ambiguity in
+     resource allocation.  
+
+2. [Invalid Pod Spec] Row 6 from the scenarios table: Pod Limits and Container
+   Limits Defined, Pod Requests and Container Requests Unspecified.
+
+   ![](./example-2.png)
+
+   * This pod will be rejected for same reasons as example 1 above (for row 5).
+     This KEP prioritizes specifying explicit resource requests at pod-level or at
+     container-level.
+
+3. Row 7 from the scenarios table: Pod Limit is set; Container Requests are set for
+   both regular containers.
+
+   ![](./example-3.png)
+
+   * Pod-level Request: Since container-level requests are set, Kubernetes can
+     determine the pod-level request using the formula from
+     [KEP#753](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/753-sidecar-containers/README.md#resources-calculation-for-scheduling-and-pod-admission).
+     As both containers in the pod are regular containers, the pod-level request
+     will be calculated as sum of container level request which is equal to
+     100Gi.
+    
+   * Container-level Limit: Container-level limit will be set equal to
+     pod-level limit only in cgroup settings (and not in PodSpec) for runtimes (or other softwares)
+     relying on container-level cgroup values.
+
+4. Pod-level Limit is set; Container Requests & Limits are set for 1 regular container.
```

Comment:
I'm not arguing to re-adjudicate at this time, but I wanted to point out that we are OK with assuming request=0 here, but we are not OK assuming request=0 when no container requests are set.

Reply:
PTAL, I updated finished updating all  the examples now.